consensus/src/consensus/request_response.rs

```diff
@@ -22,45 +27,109 @@ impl<N: Network> Consensus<N> {
         blockchain: &Arc<RwLock<Blockchain>>,
     ) {
         let stream = network.receive_requests::<RequestMacroChain>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_MACRO_CHAIN,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestBatchSet>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_BATCH_SET,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestHistoryChunk>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_HISTORY_CHUNK,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestBlock>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_BLOCK,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestMissingBlocks>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_MISSING_BLOCKS,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
 
         let stream = network.receive_requests::<RequestHead>();
-        tokio::spawn(Self::request_handler(network, stream, blockchain));
+        tokio::spawn(Self::request_handler(
+            network,
+            stream,
+            blockchain,
+            policy::MAX_RESPONSE_REQUEST_HEAD,
+            policy::MAX_RESPONSE_REQUEST_BLOCK_RANGE,
+        ));
     }
 
-    pub(crate) fn request_handler<Req: Handle<Req::Response> + Request>(
+    pub(crate) fn request_handler<Req: Handle<Req::Response, N> + Request>(
```

Comment:
the `, N` should be unnecessary as it was already specified on the `impl<N: Network>` in Line 22.
```suggestion
    pub(crate) fn request_handler<Req: Handle<Req::Response> + Request>(
```